Expand function documentation

sjspielman opened the pull request:
Closes #16 

This PR adds additional docs for the classifier functions, including more context for the parameter values we fix and how to prepare the data. I suspect there will be spots that will benefit from more precision, so please let me know where! 

Note, I pulled some context/info from this discussion: https://github.com/AlexsLemonade/mbSSP/pull/14#discussion_r2291756958 ; do you think I should also include a link to this paper for the ktsp docs?

sjspielman committed:
expand classify_ktsp docs

sjspielman committed:
start the RF update, and fix some ktsp typos

sjspielman committed:
use hlines instead of hashtags for consistency

sjspielman committed:
docs update, it's all pair-based

sjspielman committed:
render docs

sjspielman committed:
Merge branch 'main' into sjspielman/16-expand-function-docs

jaclyn-taroni commented on R/classify.R:
```diff
@@ -75,25 +94,40 @@ classify_ktsp <- function(data_mat, verbose = TRUE) {
+#' When running `multiclassPairs::predict_RF()`, all parameters are left at their
+#' defaults (although these can optionally be provided as well), except we specify
+#' `impute = TRUE`.
+#' This argument specifies to impute expression values for any missing genes, i.e.
```
```suggestion
#' This argument specifies to impute expression values for any missing genes, i.e.,
```

jaclyn-taroni commented on R/classify.R:
```diff
@@ -2,19 +2,37 @@
+#'
+#' Data preparation
+#' -------------------
+#' Because the kTSP classifier uses a pair-based approach to classify samples and operates on
```
I don't think we necessarily want to assume that people know what "pair-based approach" means here (and the same comment goes for the RF documentation).

Here's a thought:

The kTSP classifier uses rules for the relative expression of gene pairs (e.g., GeneA > GeneB) to make classifications. As such, the matrix expression values should accurately represent relative gene expression _within_ a given sample appropriately for a given technology or platform. For example, we recommend using TPM...

jaclyn-taroni reviewed (commented):
This looks mostly good. I don't think it's a good idea to assume that people will know what we mean by "pair-based approach," so we should spell it out more. I added a thought, but I'm not overly attached to my wording choice.

sjspielman committed:
Update R/classify.R

Co-authored-by: Jaclyn Taroni <19534205+jaclyn-taroni@users.noreply.github.com>

sjspielman committed:
pair-based phrasing

sjspielman committed:
caught a few spots to update while i was here

sjspielman commented:
Thanks for the feedback, let me know how this updated phrasing looks to you. I also updated phrasing for a couple spots in the `classifiers.R` docs, since I think we probably want to state pair-based there too, and I fixed a bad function name typo..!

jaclyn-taroni commented on R/classify.R:
```diff
@@ -31,7 +51,8 @@ subtype_labels <- c("WNT", "SHH", "G3", "G4")
 #' \dontrun{
-#' # Assumes input_data_matrix is a properly-formatted sample by gene expression matrix
+#' # Assumes input_data_matrix is a properly-formatted
+#' sample by gene expression matrix with relative expression values
```
Should this be? Genuinely having some trouble based on the diff
```suggestion
#' # sample by gene expression matrix with relative expression values
```

sjspielman replied:
Should be, thanks!! I'll double check these in general too.

jaclyn-taroni reviewed (approved):
LGTM

sjspielman committed:
Update R/classify.R

Co-authored-by: Jaclyn Taroni <19534205+jaclyn-taroni@users.noreply.github.com>

sjspielman committed:
redoc dontrun comments
